panel/io/document.py

```diff
@@ -208,8 +229,10 @@ def _destroy_document(self, session):
             del state_obj[self]
 
     # Schedule GC
-    at = dt.datetime.now() + dt.timedelta(seconds=5)
-    state.schedule_task('gc.collect', gc.collect, at=at)
+    global _last_cleanup
```

Comment:
```suggestion
    global _panel_last_cleanup
```

Just so nothing else changes the global `_last_cleanup`



I'm curious if we need global for this one. Is it because it is inside an event loop?

Reply:
Yep, indeed, basically sessions can be destroyed while this task is scheduled and when it runs we want to check that another session wasn't just cleaned up.